rskj-core/src/main/java/co/rsk/net/discovery/PeerExplorer.java

```diff
@@ -136,6 +146,10 @@ synchronized void start(boolean startConversation) {
     }
 
     public synchronized void dispose() {
+        if(knownPeersSaver != null) {
```

Comment:
the idea here is to trigger all dispose logic after the `state == ExecState.FINISHED` check. Similar to what `this.cleaner.dispose()` does